core/monster.py

```diff
@@ -424,7 +424,9 @@ def forward(
         disp = init_disp
         disp_preds = []
 
-        disp_up: torch.Tensor = torch.rand(7, 7)
+        # Assign a dummy tensor.
+        # Otherwise, `disp_up` is first defined inside the for-loop.
+        disp_up = torch.zeros(1)
```

Comment:
Can you do:
```suggestion
        disp_up: torch.Tensor
```
I'm guessing JIT only needs to know the type ahead of time, right?

Reply:
```
annotated assignments without assigned value aren't supported:
  File "/home/fizyr/Workspace/fizyr/python/MonSter/core/monster.py", line 429
        # Assign a dummy tensor.
        # Otherwise, `disp_up` is first defined inside the for-loop.
        disp_up: torch.Tensor
        ~ <--- HERE
    
        for itr in range(iters):
```

Note that I am using torch version 2.0.1, which is the version recommended by the authors because it is the version they used for development. I'm not aware of the changes to newer versions of torchscript.

For the specified version, leaving annotated types without an initialization is not supported.

Reply:
Ah, shame. All righty!